### ISSUE TITLE
Between statement on index is not robust enough

### ISSUE THREAD

**Issue by birdstorm** [pingcap/tispark#223]
Found by strange test result in #218. When executing where clause `between A and B` while `A > B` on index, it seems we are not handling exceptions correctly.

e.g., if we have index on real type `tp_real`,

```
select tp_real from full_data_type_table where tp_real between 4.4 and 0.5194052764001038
```

will result in failure when fetching data:

```
com.pingcap.tikv.exception.TiClientInternalException: Error reading region:
	at com.pingcap.tikv.operation.iterator.DAGIterator.readNextRegionChunks(DAGIterator.java:146)
	at com.pingcap.tikv.operation.iterator.DAGIterator.hasNext(DAGIterator.java:85)
	at scala.collection.convert.Wrappers$JIteratorWrapper.hasNext(Wrappers.scala:42)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at org.apache.spark.sql.tispark.TiHandleRDD$$anon$2.handleList$lzycompute(TiHandleRDD.scala:63)
	at org.apache.spark.sql.tispark.TiHandleRDD$$anon$2.handleList(TiHandleRDD.scala:61)
	at org.apache.spark.sql.tispark.TiHandleRDD$$anon$2.<init>(TiHandleRDD.scala:77)
	at org.apache.spark.sql.tispark.TiHandleRDD.compute(TiHandleRDD.scala:51)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:96)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
	at org.apache.spark.scheduler.Task.run(Task.scala:99)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:282)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.concurrent.ExecutionException: com.pingcap.tikv.exception.GrpcException: retry is exhausted.
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at com.pingcap.tikv.operation.iterator.DAGIterator.readNextRegionChunks(DAGIterator.java:141)
	... 26 more
Caused by: com.pingcap.tikv.exception.GrpcException: retry is exhausted.
	at com.pingcap.tikv.policy.RetryPolicy.handleFailure(RetryPolicy.java:59)
	at com.pingcap.tikv.policy.RetryPolicy.callWithRetry(RetryPolicy.java:82)
	at com.pingcap.tikv.AbstractGRPCClient.callWithRetry(AbstractGRPCClient.java:63)
	at com.pingcap.tikv.region.RegionStoreClient.coprocess(RegionStoreClient.java:217)
	at com.pingcap.tikv.operation.iterator.DAGIterator.process(DAGIterator.java:161)
	at com.pingcap.tikv.operation.iterator.DAGIterator.lambda$submitTasks$1(DAGIterator.java:59)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
Caused by: java.lang.IllegalStateException: Received other error from TiKV:unknown error Other(StringError(" is too short"))
	at com.pingcap.tikv.operation.KVErrorHandler.handle(KVErrorHandler.java:165)
	at com.pingcap.tikv.policy.RetryPolicy.callWithRetry(RetryPolicy.java:78)
	... 10 more
```

As a matter of fact, it should just return an empty set since in Spark, the sql above is equivalent to:

```
select tp_real from full_data_type_table where tp_real >= 4.4 and tp_real <= 0.5194052764001038
```

The filter result will always be false. 

Physical plan:

```
== Physical Plan ==
TiSpark RegionTaskExec{downgradeThreshold=100000,downgradeFilter=[]
+- SortAggregate(key=[RegionId#63L], functions=[CollectHandles(distinct Handles#64, 0, 0)])
   +- *Sort [RegionId#63L ASC NULLS FIRST], false, 0
      +- Exchange hashpartitioning(RegionId#63L, 200)
         +- TiDB HandleRDD{[table: full_data_type_table] [Index: idx_tp_real] , Columns: [tp_real]}
```

FYI, since TiDB is handling this correctly, it may not be a problem from TiKV but rather implementation of `DAGIterator`. I will dig into this today.

**Comment by birdstorm** [comment]
@ilovesoup PTAL

**Comment by birdstorm** [comment]
fixed by #240
